src/haddock/modules/analysis/clustfcc/__init__.py

```diff
@@ -104,10 +104,27 @@ def _run(self):
             self.params['strictness'],
             )
 
-        _, clusters = cluster_fcc.cluster_elements(
-            pool,
-            threshold=self.params['threshold'],
-            )
+        cluster_check = False
```

Comment:
I strongly suggest writing the conditionals always with positive statements.

```python
cluster_missing = True
while cluster_missing:
    for ....
          if clusters:
               cluster_missing = False
               ...
          else:
              log(...)
```

i think with this, the `if not cluster_check` in line 125 is not even needed.

Reply:
In this logic it would get stuck if no cluster is generated, this only works if `threshold=1` always generates a cluster, right?